zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/BaseVersion.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright The OpenZipkin Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package zipkin2.elasticsearch;
+
+import static zipkin2.elasticsearch.internal.JsonReaders.enterPath;
+
+import java.io.IOException;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.linecorp.armeria.common.AggregatedHttpRequest;
+import com.linecorp.armeria.common.HttpMethod;
+
+import zipkin2.elasticsearch.internal.client.HttpCall;
+
+/**
+ * Base version for both Elasticsearch and OpenSearch distributions.
+ */
+public abstract class BaseVersion {
```

Comment:
@reta so for the dependencies job, one way is to flatten this to just Version and not use a type hierarchy to retrieve the distribution property.

Then, ElasticsearchStorage.version() can return this and the depedencies main class can handle if it is "opensearch" or not (assume ES if not literally opensearch).

Reply:
only issue would be if armeria harms the dep tree of zipkin-dependencies, because right now, we only use zipkin-storage-elasticsearch as a test dep

Reply:
anyway a simple http client that gets the distribution property from json shouldn't be too bad worst case.

Reply:
> @reta so for the dependencies job, one way is to flatten this to just Version and not use a type hierarchy to retrieve the distribution property.

I was thinking about it, here is the deal: the delendencies job does not depend on zipkin-elasticsearch nor armenia, so reusing this exact approach is a bit problematic. However, we could use simple one time HTTP call instead (using HttpClient), basically just checking the distribution, not even version. Wdyt?

Reply:
> anyway a simple http client that gets the distribution property from json shouldn't be too bad worst case.

Haha, while I was typing my reply you basically came to the same conclusions 😄